spec/ddtrace/chunker_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'spec_helper'
+require 'spec/support/language_helpers'
+
+require 'ddtrace/chunker'
+
+RSpec.describe Datadog::Chunker do
+  context '.chunk_by_size' do
+    subject(:encode) { described_class.chunk_by_size(list, max_chunk_size) }
+    let(:list) { %w[1 22 333] }
+    let(:max_chunk_size) { 3 }
+
+    it do
+      expect(subject.to_a).to eq([%w[1 22], ['333']])
+    end
+
+    context 'with single element that is too large' do
+      let(:list) { ['55555'] }
+
+      it 'returns single element exceeding maximum' do
```

Comment:
Not sure I understand why this behavior is, or how we work with it?

- Are we avoiding this scenario entirely by using `encode_one` and `compact`? Rendering this moot for the trace chunking?
- Not to say we shouldn't, but is there a particular reason we should return this single element? Are there other choices we could consider as well?

Reply:
It's a matter of what happens when you ask for chunks of max_size 10, but you give it a single element of size 11?

I'm returning the oversized element here because then the caller can make an informed decision (and possibly output a message or record a metric) about this oversized element. There's no extra cost in returning the existing element, instead of discarding it, so I chose to return it and let the caller handle it.

This test asserts that this edge-case behaves the way we want.

Reply:
I see. It's a little strange, but if your only choices are return batches that are too big, or omit them entirely, I think this makes more sense.